### PR TITLE
 libsql/core: Add support for transactions

### PR DESCRIPTION
**Pull request by penberg**

### CODE REVIEW

File: crates/core/src/connection.rs [hunk 1]
```diff
@@ -103,6 +103,19 @@ impl Connection {
         }
     }
 
+    /// Begin a new transaction in DEFERRED mode, which is the default.
+    pub fn transaction(&self) -> Result<Transaction> {
+        self.transaction_with_behavior(TransactionBehavior::Deferred)
+    }
+
+    /// Begin a new transaction in the given mode.
+    pub fn transaction_with_behavior(
```
Comment (LucioFranco) [inline crates/core/src/connection.rs:112]: imho I would add a version of this fn for each enum variant and then make the enum private. Its 1) cleaner and 2) more backwards compat w/ adding new variant (which I assume in this case is likely never to happen).

File: crates/core/src/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        match self.drop_behavior {
+            DropBehavior::Rollback => {
+                self.do_rollback().unwrap();
+            }
+            DropBehavior::Commit => {
+                self.do_commit().unwrap();
+            }
+            DropBehavior::Ignore => {}
+            DropBehavior::Panic => {
+                panic!("Transaction dropped without being committed or rolled back");
```
Comment (LucioFranco) [inline crates/core/src/transaction.rs:37]: You need to check if we are panicking already https://doc.rust-lang.org/stable/std/thread/fn.panicking.html before you can panic or else a double panic is really bad.  And same with the unwraps above.

File: crates/core/src/transaction.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+impl Deref for Transaction {
+    type Target = Connection;
+
+    #[inline]
+    fn deref(&self) -> &Connection {
+        &self.conn
+    }
+}
```
Comment (LucioFranco) [inline crates/core/src/transaction.rs:91]: Why this over exposing a `as_connection` fn? iirc using deref for this is bad practice.

**Review by LucioFranco — DISMISSED** [review]

File: crates/core/src/transaction.rs (hunk 3 above)
Comment (penberg) [reply crates/core/src/transaction.rs:91]: This is just mirroring the rusqlite API.

File: crates/core/src/connection.rs (hunk 1 above)
Comment (penberg) [reply crates/core/src/connection.rs:112]: This is just mirroring the rusqlite API.

**Pull request merged**